core/io/marshalls.h

```diff
@@ -91,7 +91,7 @@ static inline unsigned int encode_half(float p_float, uint8_t *p_arr) {
 }
 
 static inline unsigned int encode_float(float p_float, uint8_t *p_arr) {
-	MarshallFloat mf;
+	MarshallFloat mf{};
```

Comment:
I'm concerned about doing explicit zeroing of the union in relatively core, possibly hot code.

But I'm also not sure what this solves. We set `f` below so the int value would also be set as it's a union, no?

I tested with gcc and clang on this code:
```cpp
#include <cstdint>
#include <iostream>

union MarshallFloat {
        uint32_t i; ///< int
        float f; ///< float
};

int main() {
	MarshallFloat mf1;
	mf1.f = 0.42f;
	std::cout << mf1.i << " " << mf1.f << std::endl;

	MarshallFloat mf2{};
	mf2.f = 0.42f;
	std::cout << mf2.i << " " << mf2.f << std::endl;

	return 0;
}
```

Both print:
```
1054280253 0.42
1054280253 0.42
```

Reply:
What was the actual report from clang-tidy?

Reply:
Ohhh boy, this one. I've tried wrangling this one a few times before. This absolutely came from `cppcoreguidelines-pro-type-member-init`, which is a warning I like in theory but **strongly** dislike `clang-tidy`'s "solutions" for. The warning occurs when the constructor leaves some members uninitialized & the declaration isn't initialized; `clang-tidy` seemingly has very little awareness of gradual initialization. Unfortunately, instead of actually suggesting fixes to the original structure/class it wants to make *any instance* of declaration zero-initialized. The "proper" solution would be:
```cpp
union MarshallFloat {
        uint32_t i; ///< int
        float f = 0; ///< float
};
```
Or, if you want only a certain type to initialize (the most "free" version):
```cpp
union MarshallFloat {
        uint32_t i; ///< int
        float f = 0; ///< float
        MarshallFloat(float p_value) :
		f(p_value) {}
};
```

Reply:
Verified locally, here's the warning output:
`uninitialized record type: 'mf' C/C++(cppcoreguidelines-pro-type-member-init)`

Sure enough, trying to "fix" the problem just zero-initializes the declaration

Reply:
Rebased to latest changes on `master` and changed to initialize one of the two values of the `union` instead of initializing each instance